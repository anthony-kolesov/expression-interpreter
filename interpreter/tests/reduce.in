var a = reduce({5, 7}, 1, x y -> x * y)
out a
print "\n"

var b = reduce({5, 7}, 0.0, left right -> left + right)
out b
print "\n"

# Use reduce result.
out a + b
print "\n"

# Map as input to reduce.
out reduce(map({1, 5}, i -> i * 2), 0, l r -> l + r) + 1000000
print "\n"

# Float input.
out reduce(map({1, 5}, i -> i * 2.0), 0, l r -> l + r)
print "\n"
