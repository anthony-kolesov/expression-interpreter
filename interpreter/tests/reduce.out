210
18.000000
228.000000
1000030
30.000000
